101-mongodb-sharded-on-centos/scripts/replica_primary.sh

```diff
@@ -0,0 +1,222 @@
+#!/bin/bash
+
+replSetName=$1
+staticIP=$2
+mongoAdminUser=$3
+mongoAdminPasswd=$4
+location=$5
+dnsNamePrefix=$6
+
+install_mongo3() {
+
+	#install
+	wget https://repo.mongodb.org/yum/redhat/7/mongodb-org/3.6/x86_64/RPMS/mongodb-org-server-3.6.17-1.el7.x86_64.rpm
+	wget https://repo.mongodb.org/yum/redhat/7/mongodb-org/3.6/x86_64/RPMS/mongodb-org-shell-3.6.17-1.el7.x86_64.rpm
+	rpm -i mongodb-org-server-3.6.17-1.el7.x86_64.rpm
+	rpm -i mongodb-org-shell-3.6.17-1.el7.x86_64.rpm
+	PATH=$PATH:/usr/bin; export PATH
+
+	#disable selinux
+	sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/sysconfig/selinux
+	sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/selinux/config
+	setenforce 0
+
+	#kernel settings
+	if [[ -f /sys/kernel/mm/transparent_hugepage/enabled ]];then
+		echo never > /sys/kernel/mm/transparent_hugepage/enabled
+	fi
+	if [[ -f /sys/kernel/mm/transparent_hugepage/defrag ]];then
+		echo never > /sys/kernel/mm/transparent_hugepage/defrag
+	fi
+
+	#configure
+	sed -i 's/\(bindIp\)/#\1/' /etc/mongod.conf
+
+	#set keyfile
+	echo "vfr4CDE1" > /etc/mongokeyfile
+	chown mongod:mongod /etc/mongokeyfile
+	chmod 600 /etc/mongokeyfile
+	sed -i 's/^#security/security/' /etc/mongod.conf
+	sed -i '/^security/akeyFile: /etc/mongokeyfile' /etc/mongod.conf
+	sed -i 's/^keyFile/  keyFile/' /etc/mongod.conf
+}
+
+
+disk_format() {
+	cd /tmp
+	for ((j=1;j<=3;j++))
+	do
+		wget https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/shared_scripts/ubuntu/vm-disk-utils-0.1.sh
+		if [[ -f /tmp/vm-disk-utils-0.1.sh ]]; then
+			bash /tmp/vm-disk-utils-0.1.sh -b /var/lib/mongo -s
+			if [[ $? -eq 0 ]]; then
+				sed -i 's/disk1//' /etc/fstab
+				umount /var/lib/mongo/disk1
+				mount /dev/md0 /var/lib/mongo
+			fi
+			break
+		else
+			echo "download vm-disk-utils-0.1.sh failed. try again."
+			continue
+		fi
+	done
+
+}
+
+yum install wget -y
+fqdn="${dnsNamePrefix}.${location}.cloudapp.azure.com"
```

Comment:
this cannot be hard-coded - it must be passed in to the script from the template and retrieved using reference()

Reply:
I have hardcoded the suffix for now since I am only supporting public cloud. Will take care of this once I add support for other clouds

Reply:
this is not optional